library/cwm/src/lib/cwm/tree.rb

```diff
@@ -0,0 +1,62 @@
+module CWM
+  # A {Tree} widget item
+  class TreeItem
+    include Enumerable
+
+    # @return what to put in Id
+    attr_reader :id
+    # @return [String]
+    attr_reader :label
+    # @return [String] icon filename
+    attr_reader :icon
+    # @return [Boolean] is the subtree open?
+    attr_reader :open
+    # @return [Hash{id => TreeItem}]
+    attr_reader :children
+
+    def initialize(id, label, icon: nil, open: true, children: [])
+      @id = id
+      @label = label
+      @icon = icon
+      @open = open
+      @children = children.map { |c| [c.id, c] }.to_h
+    end
+
+    def ui_term
+      args = [Yast::Term.new(:id, id)]
+      args << Yast::Term.new(:icon, icon) if icon
+      args << label
+      args << open
+      args << children.values.map(&:ui_term)
+      Yast::Term.new(:item, *args)
+    end
+  end
+
+  # A tree of nested {TreeItem}s
+  class Tree < CustomWidget
+    # @return [Enumerable<TreeItem>]
+    attr_reader :items
+    attr_reader :label
+
+    def initialize(* items, label:)
+      @label = label
+      @items = items
+    end
+
+    def contents
+      item_terms = items.map(&:ui_term)
+      Tree(Id(widget_id), Opt(:notify), label, item_terms)
+    end
+
+    # FIXME: CurrentBranch? item id uniqueness?
+    # TODO: extract value/value= to CurrentItemBasedWidget
+    # or declare: value_property :CurrentItem
+    def value
+      Yast::UI.QueryWidget(Id(widget_id), :CurrentItem)
+    end
+
+    def value=(val)
+      Yast::UI.ChangeWidget(Id(widget_id), :CurrentItem, val)
+    end
```

Comment:
Now I finally understand: you are proposing that the type handled by `value`, and `value=` be `TreeItem` instead of the **id** of a `TreeItem`.
Well, for my current usage (` id = @tree.value if id == @tree.widget_id `) **id** is better.